enzyme/Enzyme/FunctionUtils.cpp

```diff
@@ -1384,9 +1384,11 @@ Function *PreProcessCache::CloneFunctionWithReturns(
   F = preprocessForClone(F, topLevel);
   std::vector<Type *> RetTypes;
   if (returnValue == ReturnType::ArgsWithReturn ||
-      returnValue == ReturnType::ArgsWithTwoReturns ||
-      returnValue == ReturnType::Return ||
-      returnValue == ReturnType::TwoReturns)
+      returnValue == ReturnType::ArgsWithTwoReturns)
+    RetTypes.push_back(F->getReturnType());
+  if ((returnValue == ReturnType::Return ||
```

Comment:
The void type error here should also apply to the argswithreturn / argswithtworeturns right?

I'd suggest you move the check for void up the chain for consistency (and generally have it become an activity of return check).

Reply:
done